proto/expression.proto

```diff
@@ -165,13 +165,27 @@ enum ExprType {
 	Like = 4005;
 	RLike = 4006;
 	Case = 4007;
+
+	/* Scalar Function */
+	ScalarFunc = 10000;
+}
+
+enum ScalarFuncSig {
+	/* Math Arithmatic */
+	PlusInt = 1;
+	PlusReal = 2;
+	PlusString = 3;
+	PlusDecimal = 4;
+	PlusTime = 5;
+	PlusDuration = 6;
 }
 
 // Evaluators should implement evaluation functions for every expression type.
 message Expr {
 	optional ExprType tp = 1 [(gogoproto.nullable) = false];
 	optional bytes val = 2;
 	repeated Expr children = 3;
+	optional ScalarFuncSig sig = 4;
```

Comment:
The enum acts like a strong contract between tidb/tikv. It's easier to maintain and understand since we programmer/compiler could look and check all the signatures in the proto definitions.

It's not that clear to use `val` to store the signature name. And tidb and tikv need to maintain a map from `string` to `signature` in each codes, which could be mismatched in some time or if bug occurs.

Since all these function signatures are fixed and predefined in an organized way, it's better to use the former way, to list all the signature explicitly in the proto files.